### PR TITLE
Make Request::send more general.

### PR DESCRIPTION
**Pull request by steffahn**
Removes `+ 'static` constraint from the `impl Read` parameter.
For this, lifetime parameters are added to `Payload` and `SizedReader`.

### CODE REVIEW

**Comment by jsha** [comment]
Welcome to the repository, and thanks for the patch! I assume this is related to #202? Can you give some more background and an example of the problem you're trying to solve? I think this is probably good but I'd like to understand the problem space better.

**Comment by steffahn** [comment]
> Can you give some more background and an example of the problem you're trying to solve?

Sure, I’m coming from [this URLO thread](https://users.rust-lang.org/t/static-lifetime-with-owned-object/50637) where I originally planned on explaining to OP why the `'static` constraint is needed here. I found it _isn’t_ really needed, so I made this PR.

I guess it is related to #202 since that’s the same person as OP in the linked URLO thread.

Since there are `Read` impls in `std` for reference types like [`&'_ u8`](https://doc.rust-lang.org/std/io/trait.Read.html#impl-Read-10) or [`&'_ mut R` where `R: Read`](https://doc.rust-lang.org/std/io/trait.Read.html#impl-Read-14), I think it is a valuable addition to the API to start supporting those types.

**Review by jsha — APPROVED** [review]
Excellent, thanks for the context. That helps me understand a lot better. By the same token, we could (in a separate change) turn the `Bytes(Vec<u8>),` variant into `Bytes(&'a [u8])`, saving a copy.

This looks good to me. I'd like to wait on @algesten's review too. In the meantime, would you mind adding a short unittests that calls `send` on a reader built from ` &'_ u8`, both to demonstrate the functionality and to ensure we don't break it in the future? Thanks!

**Comment by steffahn** [comment]
I’m not sure where exactly the unit test is supposed to go.

Also, I’m going offline for today, it’s getting late here. I’ll probably check back in tomorrow. Edits by maintainers are allowed on this PR, so feel free to make a unit test yourself if you don’t want to wait for me :wink:.

**Comment by jsha** [comment]
Sounds good, have a good night!

**Review by algesten — APPROVED** [review]
This is a very good change. It's like a 2-in-1 with both fixing the 'static lifetime and the `&[u8]`.

Thanks!

**Pull request merged**
